core/client/fs/src/main/java/alluxio/client/file/DoraCacheFileSystem.java

```diff
@@ -308,24 +310,34 @@ public List<BlockLocationInfo> getBlockLocations(URIStatus status)
       throws IOException, AlluxioException {
     AlluxioURI ufsPath = convertAlluxioPathToUFSPath(new AlluxioURI(status.getUfsPath()));
     WorkerNetAddress workerNetAddress = mDoraClient.getWorkerNetAddress(ufsPath.toString());
-    // Dora does not have blocks; to apps who need block location info, we return a virtual block
-    // that has size equal to the length of the file, and located on the Dora worker which hosts
-    // the file
-    BlockLocation blockLocation = new BlockLocation().setWorkerAddress(workerNetAddress);
-    BlockInfo bi = new BlockInfo()
-        // a dummy block ID which shouldn't be used to identify the block
-        .setBlockId(1)
-        .setLength(status.getLength())
-        .setLocations(ImmutableList.of(blockLocation));
-    FileBlockInfo fbi = new FileBlockInfo()
-        .setUfsLocations(ImmutableList.of(ufsPath.toString()))
-        .setBlockInfo(bi)
-        // the block is the only block of the file, so offset is 0
-        .setOffset(0);
-    BlockLocationInfo blockLocationInfo =
-        new BlockLocationInfo(fbi, ImmutableList.of(workerNetAddress));
+    // Dora does not have blocks; to apps who need block location info, we split multiple virtual
+    // blocks from a file according to a fixed size
+    long blockSize = DEFAULT_MOCK_BLOCK_SIZE;
+    long length = status.getLength();
+    int blockNum = (int) (length / blockSize) + 1;
```

Comment:
if `length` happen to be a multiple of `blockSize`, `+ 1` would cause an extra zero-sized block. e.g. if `length = blockSize = 32MB`, there should be only one block.

Reply:
Good catch.